test/e2e/storage/mounted_volume_resize.go

```diff
@@ -107,17 +108,20 @@ var _ = utils.SIGDescribe("Mounted volume expand", func() {
 	})
 
 	It("Should verify mounted devices can be resized", func() {
-		By("Waiting for PVC to be in bound phase")
 		pvcClaims := []*v1.PersistentVolumeClaim{pvc}
-		pvs, err := framework.WaitForPVClaimBoundPhase(c, pvcClaims, framework.ClaimProvisionTimeout)
-		Expect(err).NotTo(HaveOccurred(), "Failed waiting for PVC to be bound %v", err)
-		Expect(len(pvs)).To(Equal(1))
 
-		By("Creating a deployment with the provisioned volume")
+		// The reason we use a node selector is because we do not want pod to move to different node when pod is deleted.
```

Comment:
why does the pod need to stay on the same node?

Reply:
Because test is trying to reproduce a case where volume was already mounted when resizing was attempted.  The difference is - when a new pod starts on a node, obviously global device is mounted and all that. But if you have a deployment and you delete the pod that was managed by deployment and replacement pod starts on same node, then there is no global mount happening but resize should still happen.  This e2e captures a bug we previously had in resize logic.

Reply:
There can still be timing issues I think.  Even if a pod was restarted on the same node, there might be enough time to detach and reattach to that same node.  Would a condition like this be better exercised by a unit test, where you can control the timing of events?

Reply:
Hmm possibly. In hindsight it might be better to remove this test and replace it with a table driven test - which exercises something like XFS which also reproduces this bug. However then we may not be able to run this test regularly in some environments.

Reply:
Okay, so I am thinking we should keep it for now as it is. I will add XFS tests as part of table driven test cleanup - https://github.com/kubernetes/kubernetes/issues/71105 . @msau42 wdyt?

We do need this test for now because without it, we do not have coverage for a scenario where volume could already be mounted on the node.

Reply:
I am fine with fixing this test for now, but the conditions that it requires to validate the fix is inherently flaky and not reliable.  Can you add a comment here that we should consider adding a unit test if we want to reliably test this fix?

Reply:
Added a comment here. Also opened - https://github.com/kubernetes/kubernetes/issues/74145